presto-hive/src/main/java/io/prestosql/plugin/hive/orc/OrcDeletedRows.java

```diff
@@ -235,9 +256,16 @@ private static Path createPath(String partitionLocation, DeleteDeltaLocations.De
 
         private RowId(Page page, int position)
         {
-            this.originalTransaction = BIGINT.getLong(page.getBlock(ORIGINAL_TRANSACTION_INDEX), position);
-            this.bucket = toIntExact(INTEGER.getLong(page.getBlock(BUCKET_ID_INDEX), position));
-            this.rowId = BIGINT.getLong(page.getBlock(ROW_ID_INDEX), position);
+            this(page.getChannelCount() == 1 ? -1 : BIGINT.getLong(page.getBlock(0), position),
```

Comment:
We should move this logic out of this class, and have only `public RowId(long originalTransaction, int bucket, long rowId)` constructor.